docs: the PR to end all PRs maybe (#681)
catppuccin/userstyles: docs/CONTRIBUTING.md

Reviewer: It is "a userstyle" not "an userstyle".

It is important to note that there are **two types of PRs** for this repository:

1. **New Userstyles:** Please see
   [userstyle-creation.md](./userstyle-creation.md).
2. **Other improvements:** These should follow the recommendations below, to

>    help improve the chances of a merged PR.R.

Reviewer 2: ```suggestion
   help improve the chances of a merged PR.
```

It is important to note that there are **two types of PRs** for this repository:

1. **New Userstyles:** Please see [userstyle-creation.md](./userstyle-creation.md).
2. **Other improvements:** These should follow the recommendations below, to help improve the chances of a merged PR.

## Recommendations

### Root Variables

> While writing a userstyle, you may have come across [custom properties / CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*), typically wrapped in a `:root` selector. We refer to these variables as "root variables", and they are sort of global variables used all across a website. For Catppuccin userstyles, we prefer that these variables are themed rather than individual elements. As the website likely uses these variables themselves, it saves yourself a lot of work in theming and maintaining the userstyle.

Reviewer: ```suggestion
While writing a userstyle, you may have come across [custom properties / CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*), typically wrapped in a `:root` selector. We refer to these variables as "root variables", and they can be though of as global variables used all across a website. For Catppuccin userstyles, we prefer that these variables are themed (if they exist) rather than individual elements. As the website hopefully uses these variables themselves, it saves yourself a lot of work in theming and maintaining the userstyle.
```

  file. This is to enable version control of the style.
- If it's your first time contributing to a project then you should look to the
  popular [first-contributions](https://github.com/firstcontributions/first-contributions)
  repository on GitHub. This will give you hands-on experience with the features
  of GitHub required to make a contribution. As always, feel free to join our [Discord](https://discord.com/servers/catppuccin-907385605422448742) to ask
  any questions and clarify your understanding, we are more than happy to help!
- Changes to docs may need to use [marksman](https://github.com/artempyanykh/marksman) to generate the table of contents.

## Pull Request Review Process

```mermaid
graph TD;
    A[New userstyle];
    B[Other improvements];
    A --> C[Review by userstyles staff];
    B --> D[Review by maintainer];
    C --> E{Feedback};
    D --> F{Feedback};
    E -->|Changes made| E;
    F -->|Changes made| F;
    E -->|Approved by userstyles staff| K;
    F -->|Approved by maintainer or userstyles staff| L[Merged by maintainer];
    K[Waiting period*] --> M[Merged by userstyles staff];
```

> *A waiting period is assigned only for styles that could not be tested, this
> period is usually 1–2 days.

Reviewer: ```suggestion
*A waiting period is usually started if a website is particularly complex or for any other reason could not be thoroughly/completely tested. Waiting periods typically last between 1 and 2 days.
```

It is important to note that there are **two types of PRs** for this repository:

1. **New Userstyles:** Please see
   [userstyle-creation.md](./userstyle-creation.md).
2. **Other improvements:** These should follow the recommendations below, to

>    help improve the chances of a merged PR.

Reviewer: Do we want to keep these formatting changes?

## Recommendations

### Root Variables

> Whilst writing an userstyle, its likely you have come across something like
> `--black: #000;` in the `:root` selector. These are called root variables, and
> they are used to store values that can be reused throughout the userstyle. In
> userstyles, its important that we prefer using these over theming individual
> elements, as it allows for a more consistent and maintainable userstyle.

Reviewer: ```suggestion
While writing a userstyle, you may have come across [custom properties / CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*), typically wrapped in a `:root` selector. We refer to these variables as "root variables", and they are sort of global variables used all across a website. For Catppuccin userstyles, we prefer that these variables are themed rather than individual elements. As the website likely uses these variables themselves, it saves yourself a lot of work in theming and maintaining the userstyle.
```

### Opinionated Changes

> When making changes to the userstyle, it is important to remember that different
> users would like to have different things. This is why we have a set of rules
> that we follow when making changes to the userstyle. This is why we have the
> linter to flag any changes that do not follow these rules. These rules include:
> 
> - No font changes
> - No layout changes
>   - Padding, margin, and width changes
>   - Hiding elements
>   - Changes to the position of elements

Reviewer: ```suggestion
When writing or updating a userstyle, it is important to keep in mind that different users have different preferences. To avoid lengthy discussion over user interface aesthetics, we have a set of rules for what a userstyle may include; importantly, **changes to font or layout (padding, margin, hiding/moving elements) are prohibited.**
```

### General Recommendations

- Create a
  [topic branch](https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows#_topic_branch)
  on your fork for your specific PR.
- Catppuccin uses the
  [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
  standard for creating explicit and meaningful commit messages. This repository
  requires pull request _titles_ to be in the conventional commit format,
  however we do not require it for individual commits within a pull request.
- Update the version in the `==UserStyle==` header of the `catppuccin.user.css`

>   file. This is to enable version control of the style.

Reviewer: More formatting changes... I think this is `deno fmt` at work creating these line breaks. Not a fan personally, I'll leave it up to you and Hammy though if we mind or not.

Author: I honestly don't mind, not like we see the markdown compared to the rich display.

Reviewer: Eeh I mean it makes it easier to edit...

Reviewer 2: I think prettier is going to change these anyways? I think I've seen it running over markdown :thinking:

Reviewer: > I think prettier is going to change these anyways? I think I've seen it running over markdown 🤔
> 
> Edit: Looks like it -> [b5cbfeb](https://github.com/catppuccin/userstyles/commit/b5cbfeb8a0e4d01963d7fe6a6daae035981413ce)

No, it doesn't remove forced line breaks.
